dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -455,12 +455,14 @@ public void setMetadataFromVersion(Version version) {
         this.email = version.getEmail();
     }
 
+    // This will force EAGER workflowVersions UNLESS the entry entity was detached prior to endpoint return
```

Comment:
So if you detach, it executes for the versions that are present and doesn't fetch any more? Sounds like it, just wanted to confirm.

Reply:
From what I can tell, it's lazy loaded all the way to the end of the resource endpoint function.  when it's about to serialize, it checks that it needs to get the versions (in order to figure out the output/input file formats), and then it loads it